jwst/ami/utils.py

```diff
@@ -285,3 +287,100 @@ def findmax(mag, vals, mid=1.0):
     maxx, maxy, fitc = quadratic(p, fitr)
 
     return maxx, maxy
+
+
+def pix_median_fill_value(input_array, input_dq_array, bsize, xc, yc):
+    """
+    Short Summary
+    -------------
+    For the pixel specified by (xc, yc), calculate the median value of the
+    good values within the box of size bsize neighboring pixels. If any of
+    the box is outside the data, 0 will be returned.
+
+    Parameters
+    ----------
+    input_array: ndarray
+        2D input array to filter
+    input_dq_array: ndarray
+        2D input data quality array
+    bsize: scalar
+        square box size of the data to extract
+    xc: scalar
+        x position of the data extraction
+    yc: scalar
+        y position of the data extraction
+
+    Returns
+    -------
+    median_value: float
+        median value of good values within box of neighboring pixels
+
+    """
+    # set the half box size
+    hbox = int(bsize/2)
+
+    # Extract the region of interest for the data
+    try:
+        data_array = input_array[xc - hbox:xc + hbox, yc - hbox: yc + hbox]
+        dq_array = input_dq_array[xc - hbox:xc + hbox, yc - hbox: yc + hbox]
+    except IndexError:
+        # If the box is outside the data return 0
+        log.warning('Box for median filter is outside the data.')
+        return 0.
+
+    filtered_array = data_array[(dq_array != dqflags.pixel['DO_NOT_USE']) &
+                     (np.isfinite(data_array))]
```

Comment:
You might not need to include the extra filter `np.isfinite()` if you just use `np.nanmedian()` in the next line.

Reply:
Right, and it simplifies the code.